### PR TITLE
Fix num entries scanned in filter for MultiValue scanner

### PR DESCRIPTION
**Pull request by jasperjiaguo**
The calculation of numEntriesScanned in MVScanDocIdIterator does not correctly reflect the actual number entries in multi-value case. The counter should be increase by the actual entries scanned in a multi-value instead of 1.

### CODE REVIEW

**Comment by jasperjiaguo** [comment]
@Jackie-Jiang Hi Jackie can you review this part of the code? Seems you implemented the counter. Thanks.

File: pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/MVScanDocIdIterator.java [hunk 1]
```diff
@@ -80,9 +81,11 @@ public boolean isMatch(int docId) {
       return false;
     }
     valueIterator.skipTo(docId);
-    _numEntriesScanned++;
     int length = valueIterator.nextIntVal(intArray);
-    return evaluator.applyMV(intArray, length);
+    MutableInt numEntriesScannedInEvaluator = new MutableInt(0);
+    boolean ret = evaluator.applyMV(intArray, length, numEntriesScannedInEvaluator);
+    _numEntriesScanned += numEntriesScannedInEvaluator.toInteger();
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/MVScanDocIdIterator.java]: You can make _numEntriesScanned a MutableInt and keep passing it to the evaluators. In the getNumEntriesScanned(), return _numEntriesScanned.intValue()

Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/MVScanDocIdIterator.java]: Please also fix the tests

Comment (sunithabeeram) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/MVScanDocIdIterator.java]: +1

**Comment by sunithabeeram** [comment]
It looks like the tests are failing due to timeout. @jasperjiaguo can you kick off the travis job again?

**Pull request merged**